### PR TITLE
Swap internal docker daemon spawning for reliance on host daemon

### PR DESCRIPTION
**Pull request by eeyun**
Signed-off-by: Ian Henry <ihenry@chef.io>

### CODE REVIEW

File: components/builder-worker/src/runner/docker.rs [hunk 1]
```diff
@@ -110,21 +97,17 @@ impl<'a> DockerExporter<'a> {
...
-
-        if let Some(e) = self.teardown_dockerd(dockerd).err() {
-            error!("failed to teardown dockerd instance, err={:?}", e);
-        }
+        // We used to spawn a docker daemon here but we dont anymore
+        // TODO: We should determine what broke this behavior and restore it
+        self.run_export(streamer)
 
-        exit_status
+        // We used to call a teardown of the docker daemon herer
```
Comment (chefsalim) [inline components/builder-worker/src/runner/docker.rs]: We can now merge the windows and non-windows export functions

File: components/builder-worker/src/runner/docker.rs [hunk 2]
```diff
@@ -194,12 +177,9 @@ impl<'a> DockerExporter<'a> {
         cmd.env(NONINTERACTIVE_ENVVAR, "true"); // Disables progress bars
         cmd.env("TERM", "xterm-256color"); // Emits ANSI color codes
 
-        if cfg!(not(windows)) {
-            let sock = self.dockerd_sock();
-            debug!("setting docker export command env, {}={}",
-                   DOCKER_HOST_ENVVAR, sock);
-            cmd.env(DOCKER_HOST_ENVVAR, sock); // Use the job-specific `dockerd`
-        }
+        // We use to configure a custom docker socket here for linux
```
Comment (chefsalim) [inline components/builder-worker/src/runner/docker.rs]: I don't think we need this comment

**Comment by chefsalim** [comment]
Resolves https://github.com/habitat-sh/builder/issues/1066

**Comment by eeyun** [comment]
just waiting on buildkite to pass before merge.

**Pull request merged**